### PR TITLE
CVE-2013-1896

### PR DESCRIPTION
**Pull request by Ozay34**

### CODE REVIEW

File: cves/CVE-2013-1896.yml [hunk 1]
```diff
@@ -32,7 +32,20 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  The http protocol has grown over the years, and has gained multiple extentions
+  to serve different purposes.  One such extention is WebDAV (Web Distributed
+  Authoring and Versioning) which extends the http protocol to add request methdos
+  such as COPY, MOVE, etc. to allow for remotely interacting with a servers
+  resources or documents.  One place WebDAV is used is in version controll over
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml]: nit: controll -> control

Another really good description here

File: cves/CVE-2013-1896.yml [hunk 2]
```diff
@@ -80,9 +93,12 @@ unit_tested:
...
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer:
-  code:
-  fix:
+  answer: |
+    No unit tests were added as a result of this change, this might be a good
+    thing to test for in integration testing since it involves sending certain
+    types of requests to the system as a whole.
+  code: false
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml:101]: You mentioned that no unit tests were made for the fix, was there a unit test run when the feature was initially developed?

File: cves/CVE-2013-1896.yml [hunk 3]
```diff
@@ -112,8 +131,11 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer:
-  name:
+  answer: |
+    This vulnerability resides in mod_dav, which is a file used to add version controll
+    support (or WebDAV in general) to the server. This WebDAV is essential for allowing
+    modification of files over HTTP like with the use of version controll.
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml]: nit: controll -> control

File: cves/CVE-2013-1896.yml [hunk 4]
```diff
@@ -136,7 +158,14 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer:
+  answer: |
+    The mod_dav.c file is fairly large in scope which means that many different
+    people have worked on it between the time the vulnerability was introduced
+    and when it was fixed.  Most of these changes however, were not associated
+    with the vulnerability.  Because of all of these changes however, it is
+    fairly likely that similar vulnerabilities exist in this module because most
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml:167]: Good observation

File: cves/CVE-2013-1896.yml [hunk 5]
```diff
@@ -171,8 +204,11 @@ lessons:
     applies:
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+      Anything can be passed in the body of an HTTP request, and one of the most
+      important features of an HTTP server is to ensure that the infomormation is
+      sanatized, or filtered to ensure that only real requests are made.
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml:214]: You should also mention that servers should check for non-native HTTP requests (like those part of WebDAV) and find a way to handle such requests

File: cves/CVE-2013-1896.yml [hunk 6]
```diff
@@ -204,4 +245,12 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    Although at it's core, the patch for this was just originally a coding mistake
+    placed in the wrong function, overall, this is an architectural design mistake.
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml:255]: Agreed

File: cves/CVE-2013-1896.yml [hunk 7]
```diff
@@ -189,8 +225,13 @@ lessons:
     applies:
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      Version controll is a very complex system with many different ways to interact.
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml]: Control is spelled incorrectly again

File: cves/CVE-2013-1896.yml [hunk 8]
```diff
@@ -99,11 +115,14 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer:
-  date:
-  automated:
-  google:
-  contest:
+  answer: |
+    The vulnerability was discovered and a fix was proposed (and accepted) by
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml]: Maybe some clarification could bse useful here. Right now, it sounds like the same guy who discovered the fix also wrote the fix and then accepted his own fix(?) Probably not the case, but that's how the sentence reads.

File: cves/CVE-2013-1896.yml [hunk 9]
```diff
@@ -159,8 +188,12 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies:
-    note:
+    applies: true
+    note: |
+      There are multiple layers of security when it comes to http requests,
```
Comment (exl5325) [inline cves/CVE-2013-1896.yml:194]: This sounds like something that should go in `distrust input` rather than `defense in depth`

**Review by exl5325 — CHANGES_REQUESTED** [review]

File: cves/CVE-2013-1896.yml (hunk 9 above)
Comment (Ozay34) [reply cves/CVE-2013-1896.yml:194]: I think it applies to both here, although it is true that you should not trust the input from an HTTP request, you should also provide multiple layers of security when validating an HTTP request so there are multiple walls to block potentially harmful input like what we see in this vulnerability.

File: cves/CVE-2013-1896.yml (hunk 2 above)
Comment (Ozay34) [reply cves/CVE-2013-1896.yml:101]: Added a bit of clarification

File: cves/CVE-2013-1896.yml (hunk 5 above)
Comment (Ozay34) [reply cves/CVE-2013-1896.yml:214]: Added a bit more about non-native HTTP requests

File: cves/CVE-2013-1896.yml (hunk 8 above)
Comment (Ozay34) [reply cves/CVE-2013-1896.yml]: Reworded to make this less confusing

**Pull request merged**